pkg/policies/action/action.go

```diff
@@ -0,0 +1,757 @@
+				Str("org", owner).
+				Str("repo", repo).
+				Str("area", polName).
+				Str("action", a.name).
+				Errs("errors", errors).
+				Msg("Errors while evaluating deny rule.")
+		}
+
+		results = append(results, denyResult)
+	}
+
+	// => Next, evaluate require rules
+
+	var wfr *github.WorkflowRuns
+	var headSHA string
+
+	for _, r := range applicableRules {
+		if r.Method == "require" {
+			if r.MustPass && wfr == nil {
+				var err error
+				hash, err := getLatestCommitHash(ctx, c, owner, repo)
+				if err != nil {
+					log.Error().
+						Str("org", owner).
+						Str("repo", repo).
+						Str("area", polName).
+						Err(err).
+						Msg("Error getting latest commit hash. Skipping rule evaluation.")
+					break
+				}
+				headSHA = hash
+			}
+
+			result, err := evaluateRequireRule(ctx, c, owner, repo, r, actions, headSHA, gc, sc)
+			if err != nil {
+				log.Warn().
+					Str("org", owner).
+					Str("repo", repo).
+					Str("area", polName).
+					Err(err).
+					Msg("Error evaluating require rule")
+				continue
+			}
+			results = append(results, result)
+		}
+	}
+
+	d := details{}
+
+	passing := true
+	combinedExplain := ""
+
+	// Use this map to dedupe Rules
+	failedRules := map[*Rule]struct{}{}
+
+	for _, result := range results {
+		if !result.passed() {
+			passing = false
+			if combinedExplain != "" {
+				combinedExplain += "\n"
+			}
+			combinedExplain += result.explain()
+			failedRules[result.relevantRule()] = struct{}{}
+		}
+	}
+
+	for r, _ := range failedRules {
+		d.FailedRules = append(d.FailedRules, r)
+	}
+
+	notifyText := fmt.Sprintf(failText, combinedExplain, polName)
+
+	if passing {
+		notifyText = "OK"
+	}
+
+	return &policydef.Result{
+		Enabled:    enabled,
+		Pass:       passing,
+		NotifyText: notifyText,
+		Details:    d,
+	}, nil
+}
+
+// Fix implementing policydef.Policy.Fix(). Not supported.
+func (a Action) Fix(ctx context.Context, c *github.Client, owner, repo string) error {
+	log.Warn().
+		Str("org", owner).
+		Str("repo", repo).
+		Str("area", polName).
+		Msg("Action fix is configured, but not implemented.")
+	return nil
+}
+
+// GetAction returns the configured action from Action Use policy's
+// configuration stored in the org repo, default log. Implementing
+// policydef.Policy.GetAction()
+func (a Action) GetAction(ctx context.Context, c *github.Client, owner, repo string) string {
+	oc := getConfig(ctx, c, owner, repo)
+	return oc.Action
+}
+
+func getConfig(ctx context.Context, c *github.Client, owner, repo string) *OrgConfig {
+	oc := &OrgConfig{ // Fill out non-zero defaults
+		Action: "log",
+	}
+	if err := configFetchConfig(ctx, c, owner, "", configFile, config.OrgLevel, oc); err != nil {
+		log.Error().
+			Str("org", owner).
+			Str("repo", repo).
+			Str("configLevel", "orgLevel").
+			Str("area", polName).
+			Str("file", configFile).
+			Err(err).
+			Msg("Unexpected config error, using defaults.")
+	}
+	// Initialize values in each rule
+	for _, g := range oc.Groups {
```

Comment:
Continuing the above comment, here we would convert the external org config to an internal config (similar to merged config in other policies) that contains the computed values for priority and link back to group.